pkg/reconciler/workload/virtualworkspaceurls/virtualworkspaceurls_reconcile.go

```diff
@@ -34,6 +34,11 @@ import (
 func (c *Controller) reconcile(syncTarget *workloadv1alpha1.SyncTarget, workspaceShards []*v1alpha1.ClusterWorkspaceShard) (*workloadv1alpha1.SyncTarget, error) {
 	syncTargetCopy := syncTarget.DeepCopy()
 
+	syncTargetKey := workloadv1alpha1.ToSyncTargetKey(logicalcluster.From(syncTargetCopy), syncTargetCopy.Name)
+	if val, ok := syncTargetCopy.Labels[workloadv1alpha1.InternalSyncTargetKeyLabel]; !ok || val != syncTargetKey {
+		syncTargetCopy.Labels[workloadv1alpha1.InternalSyncTargetKeyLabel] = syncTargetKey
```

Comment:
just unconditionally set it. No need for the 2 lines above.

Reply:
done